assets/js/log-viewer/store.ts

```diff
@@ -66,7 +66,12 @@ function coerceLogs(logs: LogLine[]): LogLine[] {
 }
 
 function formatLogLine(log: LogLine) {
-  return `${log.source} ${log.message}`;
+  try {
+    const jsonObj = JSON.parse(log.message);
```

Comment:
This is untested code, but I think this approach will work a little better:
```
if (typeof log.message[0] === 'string') {
  return `${log.source} ${log.message}`;
}
return `${log.source} ${JSON.stringify(jsonObj, null, 2)}`;
```

Reply:
That will only work if the first item is a string. So `console.log('result', state.data)` won't pretty print.

Message is always an array of strings or JSON (I think??).

I suppose either way we might need to iterate over the message array, pretty print any objects, and then log. Which is a little more complicated

Reply:
the
```
if (typeof log.message[0] === 'string') {
  return `${log.source} ${log.message}`;
}
```
doesn't seem to work for me because we _do_ want to try it as JSON if it's a string. worker is always sending strings or arrays of strings, and it's up to us to give them the old college try and see if we can represent them as beautiful JSON

Reply:
huh, are you sure the worker isn't sending objects?

I might have to go and look into that. My expectation is that a log event is a an object with `{ source, message }` etc keys, and message is always an array of strings (if actual strings) or objects (if actual objects). Like it shouldn't be double encoding them.

Reply:
yeah peep this

<img width="1942" alt="image" src="https://github.com/OpenFn/lightning/assets/8732845/c4dc5056-8ce4-43eb-b9ad-16bda5f640a6">

seems to always be a string?

Reply:
Yes, we take whatever the worker says and turn it into a string: https://github.com/OpenFn/lightning/blob/main/lib/lightning/ecto_types.ex#L63

Reply:
I have always intended to come back to this and possible change the message to be either an object or a string... but for the sake of moving forward (when we first implemented this) we opted for storing everything as a string.

Reply:
Oh interesting.

Alright, well for the time forcing a JSON.parse is the way to go.

I think in the long run JSON should be be send natively down to the client, which can render them as collapsible/formatted JSON in monaco. But that can all be far future stuff.